lib/ain-ocean/src/data_acces/masternode_states.rs

```diff
@@ -13,25 +14,71 @@ pub struct MasterStatsDb {
+    pub async fn get_latest(&self) -> Result<Option<MasternodeStats>> {
+        let latest_height_bytes = match self
+            .db
+            .get("masternode_block_height", b"master_block_height")
+        {
+            Ok(Some(value)) => value,
+            Ok(None) => return Ok(None), // No latest block height set
+            Err(e) => return Err(anyhow!(e)),
+        };
 
-        // let iter = self.db.iterator("masternode_stats", IteratorMode::End); // Start from the end of the DB
+        // Convert the latest height bytes back to an integer
+        let latest_height = i32::from_be_bytes(
+            latest_height_bytes
+                .as_slice()
+                .try_into()
+                .map_err(|_| anyhow!("Byte length mismatch for latest height"))?,
+        );
 
-        // for (key, value) in iter {
-        //     let stats: MasternodeStats = serde_json::from_slice(&value)?;
-        //     if stats.block.height > highest_height {
-        //         highest_height = stats.block.height;
-        //         latest_stats = Some(stats);
-        //     }
-        // }
-
-        // Ok(latest_stats);
-        todo!()
+        // Retrieve the block with the latest height
+        match self.db.get("block", &latest_height.to_be_bytes()) {
+            Ok(Some(value)) => {
+                let block: MasternodeStats =
+                    serde_json::from_slice(&value).map_err(|e| anyhow!(e))?;
+                Ok(Some(block))
+            }
+            Ok(None) => Ok(None), // No block found for the latest height
+            Err(e) => Err(anyhow!(e)),
+        }
     }
-    pub async fn query(&self, limit: i32, lt: i32) -> Result<Vec<MasternodeStats>> {
-        todo!()
+    pub async fn query(
+        &self,
+        limit: i32,
+        lt: i32,
+        sort_order: SortOrder,
+    ) -> Result<Vec<MasternodeStats>> {
+        let iterator = self.db.iterator("masternode_stats", IteratorMode::End)?;
+        let mut master_node: Vec<MasternodeStats> = Vec::new();
+        let collected_blocks: Vec<_> = iterator.collect();
+
+        for result in collected_blocks.into_iter().rev() {
+            let (key, value) = match result {
+                Ok((key, value)) => (key, value),
+                Err(err) => return Err(anyhow!("Error during iteration: {}", err)),
+            };
+
+            let master_stats: MasternodeStats = serde_json::from_slice(&value)?;
+
+            if master_stats.block.height < lt {
+                master_node.push(master_stats);
+
+                if master_node.len() == limit as usize {
+                    break;
+                }
+            }
+        }
+
+        // Sort blocks based on the specified sort order
+        match sort_order {
+            SortOrder::Ascending => master_node.sort_by(|a, b| a.block.height.cmp(&b.block.height)),
+            SortOrder::Descending => {
+                master_node.sort_by(|a, b| b.block.height.cmp(&a.block.height))
+            }
+        }
```

Comment:
Once we convert the `lt` and `sort_order` to rocksdb iterator mode, most of the list logic can be handled with something such as :
```
self.db
            .iterator(<cf_name>, iter_mode)?
            .into_iter()
            .take(limit as usize)
            .map(|result| {
                result
                    .map_err(|e| anyhow!("Error during iteration: {}", e))
                    .and_then(|(_key, value)| serde_json::from_slice(&value).map_err(Into::into))
            })
            .collect()
```

This cleans it up and removes the need for a mut vector and sorting post collecting

Reply:
Can apply to other query impl as well